ui/apps/platform/src/Components/CompoundSearchFilter/components/CompoundSearchFilter.tsx

```diff
@@ -107,18 +117,14 @@ function CompoundSearchFilter({
                 selectedAttribute={currentAttribute}
                 onChange={(value) => {
                     setSelectedAttribute(ensureString(value));
-                    setInputValue('');
                 }}
                 config={config}
             />
             {entity && attribute && (
                 <CompoundSearchFilterInputField
+                    key={getEntityAttributeKey(entity, attribute)}
```

Comment:
Is the `key` prop here to force React to discard all local component state when those values change?